scripts/build/Jenkins/Jenkinsfile

```diff
@@ -428,20 +428,33 @@ def ExportTestScreenshots(Map options, String branchName, String platformName, S
         }
     }
 }
-def ArchiveArtifactsOnS3(Map options, String artifactsSourceDir, String artifactsRegex, String s3Prefix) {
-    def command = "${options.PYTHON_DIR}/python"
-    if(env.IS_UNIX) command += '.sh'
-    else command += '.cmd'
-    command += " -u scripts/build/tools/copy_file.py -s ${artifactsSourceDir} -r \"${artifactsRegex}\" -t ${s3Prefix}"
-    palSh(command, "Copying artifacts")
-    archiveArtifacts artifacts: "${s3Prefix}/**"
+
+// All files are included by default.
+// --include will only re-include files that have been excluded from an --exclude filter.
+//See more details at https://docs.aws.amazon.com/cli/latest/reference/s3/#use-of-exclude-and-include-filters
+def ArchiveArtifactsOnS3(String artifactsSourceDir, String s3Prefix="", boolean recursive=false, List<String> includes=[], List<String> excludes=[]) {
+    if (!fileExists(s3Prefix)) {
+        palMkdir(s3Prefix)
+    }
+    palSh("echo ${env.BUILD_URL} > ${s3Prefix}/build_url.txt")
+    // archiveArtifacts is very slow, so we only archive one file and upload the rest artifacts to the same bucket using S3 CLI.
+    archiveArtifacts artifacts: "${s3Prefix}/build_url.txt"
+    def command = "aws s3 cp ${artifactsSourceDir} s3://${env.JENKINS_ARTIFACTS_S3_BUCKET}/${env.JENKINS_JOB_NAME}/${env.BUILD_NUMBER}/artifacts/${s3Prefix} "
+    includes.each{ include ->
+        command += "--include \"${include}\" "
+    }
+    excludes.each{ exclude ->
+        command += "--exclude \"${exclude}\" "
+    }
```

Comment:
I think these needs to be swapped, so that --exclude goes first.

Reply:
Good point! I missed this part, will update